src/Common/src/System/Security/Cryptography/DSACng.ImportExport.cs

```diff
@@ -0,0 +1,365 @@
+                }
+                else
+                {
+                    GenerateV2DsaBlob(out blob, parameters, keySizeInBytes, hasPrivateKey);
+                }
+
+                ImportKeyBlob(blob, hasPrivateKey);
+            }
+
+            private static void GenerateV1DsaBlob(out byte[] blob, DSAParameters parameters, int cbKey, bool includePrivate)
+            {
+                // We need to build a key blob structured as follows:
+                //
+                //     BCRYPT_DSA_KEY_BLOB       header
+                //     byte[cbKey]               P
+                //     byte[cbKey]               G
+                //     byte[cbKey]               Y
+                //     -- Private only --
+                //     byte[Sha1HashOutputSize]  X
+
+                unsafe
+                {
+                    int blobSize = 
+                        sizeof(BCRYPT_DSA_KEY_BLOB) +
+                        cbKey +
+                        cbKey +
+                        cbKey;
+
+                    if (includePrivate)
+                    {
+                        blobSize += Sha1HashOutputSize;
+                    }
+
+                    blob = new byte[blobSize];
+                    fixed (byte* pDsaBlob = blob)
+                    {
+                        // Build the header
+                        BCRYPT_DSA_KEY_BLOB* pBcryptBlob = (BCRYPT_DSA_KEY_BLOB*)pDsaBlob;
+
+                        pBcryptBlob->Magic = includePrivate ? KeyBlobMagicNumber.BCRYPT_DSA_PRIVATE_MAGIC : KeyBlobMagicNumber.BCRYPT_DSA_PUBLIC_MAGIC;
+                        pBcryptBlob->cbKey = cbKey;
+
+                        int offset = sizeof(KeyBlobMagicNumber) + sizeof(int); // skip Magic and cbKey
+
+                        if (parameters.Seed != null)
+                        {
+                            // The Seed length is hardcoded into BCRYPT_DSA_KEY_BLOB, so check it now we can give a nicer error message.
+                            if (parameters.Seed.Length != Sha1HashOutputSize)
+                                throw new ArgumentException(SR.Cryptography_InvalidDsaParameters_SeedRestriction_ShortKey);
+
+                            Interop.BCrypt.EmitBigEndian(blob, ref offset, parameters.Counter);
+                            Interop.BCrypt.Emit(blob, ref offset, parameters.Seed);
+                        }
+                        else
+                        {
+                            // If Seed is not present, back fill both counter and seed with 0xff. Do not use parameters.Counter as CNG is more strict than CAPI and will reject
+                            // anything other than 0xffffffff. That could complicate efforts to switch usage of DSACryptoServiceProvider to DSACng.
+                            for (int i = 0; i < Sha1HashOutputSize + sizeof(int); i++)
+                            {
+                                Interop.BCrypt.EmitByte(blob, ref offset, 0xff);
```

Comment:
Nit: This is going to call `EmitByte` 24 times. Instead, maybe `EmitByte` can have a count parameter of the number of times to emit the byte value, e.g. something like:

```c#
internal static void EmitByte(byte[] blob, ref int offset, byte value, int count)
{
    Debug.Assert(blob != null);
    Debug.Assert(offset >= 0);
    Debug.Assert(count > 0);

    int finalOffset = offset + count;
    for (int i = offset; i < finalOffset; i++)
    {
        blob[i] = value;
    }
    offset = finalOffset;
}
```

And then call it like:

```c#
Interop.BCrypt.EmitByte(blob, ref offset, value: 0xff, count: Sha1HashOutputSize + sizeof(int));
```